### PR TITLE
impr(SEO): Semantic HTML (borisnezlobin)

### LINKED ISSUE #4857: HTML tags aren't entirely semantic
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

Yes

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes

### Account name

randomletters

### Account config

_No response_

### Current Behavior

The HTML tags throughout the frontend are lacking in semantic structure (i.e., h1 for titles, h2 for subtitles, ul/ol for lists, etc)

### Expected Behavior

Currently, the layout looks something like:
``` 
 <div class="section">
    <div class="bigtitle">about</div>
    <h2>
      Monkeytype is a minimalistic and customizable typing test. It features
      many test modes, an account system to save your typing speed history, and
      user-configurable features such as themes, sounds, a smooth caret, and
      more. Monkeytype attempts to emulate the experience of natural keyboard
      typing during a typing test, by unobtrusively presenting the text prompts
      and displaying typed characters in-place, providing straightforward,
      real-time feedback on typos, speed, and accuracy.
      <br />
      <br />
      Test yourself in various modes, track your progress and improve your
      speed.
    </h2>
  </div>
```
Using semantics, it would look like:
```
  <section class="section">
    <h1 class="bigtitle">about</h1>
    <p>
      Monkeytype is a minimalistic and customizable typing test. It features
      many test modes, an account system to save your typing speed history, and
      user-configurable features such as themes, sounds, a smooth caret, and
      more. Monkeytype attempts to emulate the experience of natural keyboard
      typing during a typing test, by unobtrusively presenting the text prompts
      and displaying typed characters in-place, providing straightforward,
      real-time feedback on typos, speed, and accuracy.
      <br />
      <br />
      Test yourself in various modes, track your progress and improve your
      speed.
    </p>
  </section>
```
This would improve SEO and make things more clear for developers

### Steps To Reproduce

Go to any .html file in the frontend

### Environment

- OS: all
- Browser: all
- Browser Version: all

### Anything else?

https://www.w3schools.com/html/html5_semantic_elements.asp

Can I work on this? It's rather tedious, but worth doing

### PR DESCRIPTION
**Pull request by borisnezlobin**
### Description
Changes the HTML to be more semantic (e.g., changing `<div class="section">` to just `<section>`)
This PR currently only changes one page (`/about`) in order to get progress feedback before changes are made to everything.

### Checks
No visual changes
- [x] Adding a language or a theme?
- [x] Check if any open issues are related to this PR; if so, be sure to tag them below.
    - #4857 
- [x] Make sure the PR title follows the Conventional Commits standard. (https://www.conventionalcommits.org for more info)
- [x] Make sure to include your GitHub username inside parentheses at the end of the PR title

Closes #4857

### CODE REVIEW

File: frontend/static/html/pages/about.html [hunk 1]
```diff
@@ -49,10 +49,10 @@
       <canvas id="publicStatsHistogramChart"></canvas>
     </div>
     <p class="small">distribution of time 60 leaderboard results (wpm)</p>
-  </div>
-  <div class="section">
-    <div class="bigtitle">about</div>
-    <h2>
+  </section>
+  <section>
+    <h1 class="bigtitle">about</h1>
```
Comment (Miodec) [inline frontend/static/html/pages/about.html:54]: Use h2 here, then you can remove the bigtitle class

File: frontend/static/html/pages/about.html [hunk 2]
```diff
@@ -64,18 +64,18 @@ <h2>
       <br />
       Test yourself in various modes, track your progress and improve your
       speed.
-    </h2>
-  </div>
-  <div class="section">
+    </p>
+  </section>
+  <section>
     <div class="title">word set</div>
```
Comment (Miodec) [inline frontend/static/html/pages/about.html:70]: Use h3, then remove the title class

**Review by Miodec — CHANGES_REQUESTED** [review]

File: frontend/static/html/pages/about.html (hunk 2 above)
Comment (Miodec) [reply frontend/static/html/pages/about.html:70]: Also use h3 everywhere where you already used h2

**Comment by borisnezlobin** [comment]
Will do!
should I apply these changes everywhere else then?

**Comment by Miodec** [comment]
Everywhere else you mean other pages?

**Comment by borisnezlobin** [comment]
yep!

**Comment by Miodec** [comment]
Yeah sure

**Comment by fehmer** [comment]
hi @borisnezlobin, 
can you add ids to all headers so they can be used as anchors in links?

**Comment by borisnezlobin** [comment]
Will do! I have finals this week, but will be able to work on this later this week + next

**Pull request closed without merging**